apps/web/components/apps/AppPage.tsx

```diff
@@ -142,10 +144,13 @@ export const AppPage = ({
       const data = appDbQuery.data;
 
       const credentialsCount = data?.credentials.length || 0;
-      setShowDisconnectIntegration(
-        data?.userAdminTeams.length ? credentialsCount >= data?.userAdminTeams.length : credentialsCount > 0
-      );
-      setExistingCredentials(data?.credentials.map((credential) => credential.id) || []);
+      setExistingCredentials(data?.credentials || []);
+
+      const appInstalled =
+        enabledOnTeams && data?.userAdminTeams
+          ? data?.userAdminTeams.length < credentialsCount
+          : credentialsCount > 0;
```

Comment:
This conditional seems to be the same logic as earlier except for an additional check. I think we shouldn't change the approach unnecessarily here, makes it difficult to review code

Reply:
Infact it is missing the equal case. So, if the app is installed for all teams then `data?.userAdminTeams.length == credentialsCount` is true and in that case app, still shows Install button.

Reply:
Fixed it.